pkg/webhook/mutation/pod/metadata/workload.go

```diff
@@ -102,6 +102,10 @@ func findRootOwner(ctx context.Context, clt client.Client, partialObjectMetadata
 		}
 	}
 
+	if partialObjectMetadata.ObjectMeta.Name == "" {
```

Comment:
I think this would be done better after the whole recursion is done. Move this to ~ line 60.

Reply:
thanks for pointing out, you are right. done in https://github.com/Dynatrace/dynatrace-operator/pull/3170/commits/8fe5b61c123083cd2161db67f9379d4c06a58b68